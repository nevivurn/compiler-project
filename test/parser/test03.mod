//
// test03
//
// parser test:
// - integer expressions
//

module test03;

var a   : integer;
    b,c : integer;

begin
  a := a + b + c;
  b := a - b - c;
  c := a + b * c;
  b := (a + b) / c
end test03.
