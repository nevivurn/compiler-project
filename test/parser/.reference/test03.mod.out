parsing 'test03.mod'...
successfully parsed.
  AST:
    CAstScope: 'test03'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ @a        <integer>           a           ]
          [ @b        <integer>           b           ]
          [ @c        <integer>           c           ]
          [ main     <NULL>           ]
        ]]
      statement list:
        := <integer>
          [ @a        <integer>           a ] <integer>
          add <integer>
            add <integer>
              [ @a        <integer>           a ] <integer>
              [ @b        <integer>           b ] <integer>
            [ @c        <integer>           c ] <integer>
        := <integer>
          [ @b        <integer>           b ] <integer>
          sub <integer>
            sub <integer>
              [ @a        <integer>           a ] <integer>
              [ @b        <integer>           b ] <integer>
            [ @c        <integer>           c ] <integer>
        := <integer>
          [ @c        <integer>           c ] <integer>
          add <integer>
            [ @a        <integer>           a ] <integer>
            mul <integer>
              [ @b        <integer>           b ] <integer>
              [ @c        <integer>           c ] <integer>
        := <integer>
          [ @b        <integer>           b ] <integer>
          div <integer>
            add <integer>
              [ @a        <integer>           a ] <integer>
              [ @b        <integer>           b ] <integer>
            [ @c        <integer>           c ] <integer>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest03.mod.ast.pdf test03.mod.ast.dot


Done.
